Extension/src/LanguageServer/ui.ts

```diff
@@ -290,14 +289,15 @@ export class UI {
         };
 
         if (this.configurationUIPromise) {
-            this.configurationUIPromise = this.configurationUIPromise.then(result => {
+            const result: ConfigurationResult = await this.configurationUIPromise;
+            this.configurationUIPromise = new Promise<ConfigurationResult>((resolve, reject) => {
```

Comment:
Why is a new Promise being added here?  There is no async within its contents, so making a promise out of it doesn't seem to accomplish anything.  It was previously a promise in order to get chained(?) after the previous configurationUIPromise, but that was awaited on the previous line.  Since it was used for chaining, I'm not sure if an await on the previous one is the right thing to do.  You may want to talk to Michelle, who I think wrote the code in this area.  She was using promise chaining to make some functionality serial.  It's unclear to me if configurationUIPromise will be needed after your changes.

Reply:
@michelleangela can you take a look at the latest change?

Reply:
Update: working with Michelle on this